src/app/store/general/utils/powerTypes.ts

```diff
@@ -56,12 +56,32 @@ const getPowerFieldSchema = (fieldType: PowerFieldType) => {
     case PowerFieldType.MULTIPLE_CHOICE:
       return Yup.array().of(Yup.string());
     case PowerFieldType.IP_ADDRESS:
-    case PowerFieldType.VIRSH_ADDRESS:
-    case PowerFieldType.LXD_ADDRESS:
       return Yup.string().test({
         name: "is-ip-address",
         message: "Please enter a valid IP address.",
-        test: (value) => isIP(value as string),
+        test: (value) => {
+          if (typeof value !== "string") {
+            return false;
+          }
+          if (value.includes("[") && value.includes("]")) {
+            // This is an IPv6 address with a port number
+            const openingBracketIndex = value.indexOf("[");
+            const closingBracketIndex = value.indexOf("]");
+            const ip = value.slice(
+              openingBracketIndex + 1,
+              closingBracketIndex
+            );
+            // We use +2 here to include the `:` before the port number
+            const port = value.slice(closingBracketIndex + 2);
+            return isIPv6(ip) && !isNaN(parseInt(port));
```

Comment:
This usually works but insane users like myself might try: `[aa:cf:f::34fF]:  4 5 6`. Results in a `parseInt("  4 5 6")` which is `4`. I think it's still OK to merge for now but we might want to make this even tighter later.

Reply:
That's interesting! By the looks of it, `parseInt` ignores any leading whitespace, and ignores anything once it reaches the first whitespace after a number (only if the number is also the first non-whitespace text in the string).

I guess the simplest check for this would just be to check for any whitespace in the input, and treat the value as invalid if there is whitespace.